docs/technical-overview.md

```diff
@@ -0,0 +1,64 @@
+[Back to overview](./README.md)
+
+# Technical Overview
+
+The Plugin Checker works by executing checks against a single plugin. Each check will test for a specific issue and raise either an error or a warning depending on the severity.
+
+The Plugin Checker performs these checks by following the process below:
+
+1. Determine which checks to run based on the request.
+2. Execute necessary check preparations to set up the environment.
+3. Execute a comprehensive set of checks against the plugin.
+4. Ensure all errors and warnings raised during the checks are reported.
+
+
+## Checks
+Checks are individual PHP classes designed to test a plugin for a specific problem when executed.
+
+Each check contains a `run()` method that accepts an instance of the `Check_Result` class. This `Check_Result` instance is used to collect and report any errors or warnings raised by the check.
+
+The Plugin Checker currently supports two types of checks: static and runtime checks.
+
+### Static Checks
+
+Static checks are used to perform tests against the codebase without running any code, similar to other static analysis tools like PHP Code Sniffer.
+
+Static checks can utilize existing PHPCodeSniffer sniffs, such as those in the [WordPress Coding Standards](https://github.com/WordPress/WordPress-Coding-Standards), or they can be implemented manually by searching for specific patterns across all files, similar to the checks in the [WordPress Theme Check tool](https://github.com/WordPress/theme-check).
+
+Static checks implement the `Static_Check` interface.
+
+### Runtime Checks
+
+Runtime checks perform tests by executing the plugin code within a dedicated test environment, allowing for verification of specific outputs.
+
+Runtime checks execute code within a test environment, often requiring preparations to configure the runtime environment with the necessary configuration prior to performing the checks.
+
+Runtime checks implement the `Runtime_Check` interface.
+
+### Check Stabilities
+
+Each check is assigned a stability value to ensure appropriate check selection based on specific use cases. Currently, there are two stability values: Stable and Experimental.
+
+Checks are assigned a stability value using either the `Stable_Check` or `Experimental_Check` trait. Those with the `Stable_Check` trait are always executed on both the CLI and WordPress admin screens, whereas Experimental checks are exclusively run via the CLI using the `--include-experimental` flag.
+
+### Check Categories
+
+Check categories enable the execution of checks for specific use cases. Each check requires a category, which is determined by implementing a `get_categories` method within the check class. This method returns an array containing one or more categories to which the check belongs. The categories should be specified using the constants found in the `Check_Categories` class, rather than setting them as strings.
```

Comment:
This is more of a comment/observation than a change request: It is stated that one should only use the constants found in the `Check_Categories` class. Is there (or should there be) a means of extending the categories? If so, we should reference the filters. If not, it MIGHT be worth stating that custom categories are not (yet??) supported and adding a category that is not one of these constants, will have no effect and the check won't appear?

Reply:
Thank you, @10upsimon, for your question. Currently, our plugin supports only the categories that are listed here: [Check_Categories.php](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Check_Categories.php#L17-L22). However, users can still define their own categories when registering new checks. It is important to note that, as of now, these custom categories may not be reflected in the WP admin screen. Nevertheless, users can make use of these custom categories when employing the CLI command to filter results by category if desired. We do plan to implement a filter for extending categories in the future, as and when it becomes necessary.

Reply:
@mukeshpanchal27 I think it's important to note these important points in the docs.